libunsf.c

```diff
@@ -3324,7 +3339,7 @@ UNSF_SYMBOL void unsf_convert_sf_to_gus(UnSF_Options *options) {
                                     if ((sf_num_instrument_indexes * 4 != subchunk.size) ||
                                         (sf_instrument_indexes)) BAD_SF();
 
-                                    sf_instrument_indexes = malloc(sizeof(sfInstBag) * sf_num_instrument_indexes);
+                                    sf_instrument_indexes = (sfInstBag *) malloc(sizeof(sfInstBag) * sf_num_instrument_indexes);
```

Comment:
Why do we need to cast?  

* It is unnecessary, as void * is automatically and safely promoted to any other pointer type in this case.
* It can hide an error, if you forgot to include <stdlib.h>. This can cause crashes (or, worse, not cause a crash until way later in some totally different part of the code). Consider what happens if pointers and integers are differently sized; then you're hiding a warning by casting and might lose bits of your returned address.
* It adds clutter to the code, casts are not very easy to read (especially if the pointer type is long).
* It makes you repeat yourself, which is generally bad.

Is there some overriding reason why we 'need' to do this?